File: service/src/main/java/net/huray/backend/minuting/web/home/MinutesController.java
package net.huray.backend.minuting.web.home;

import lombok.RequiredArgsConstructor;
import net.huray.backend.minuting.contract.MinutesContract;
import net.huray.backend.minuting.dto.MinutesDto;
import net.huray.backend.minuting.service.MinutesReadService;
import net.huray.backend.minuting.service.MinutesWriteService;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
public class MinutesController implements MinutesContract {

    private final MinutesWriteService minutesWriteService;
    private final MinutesReadService minutesReadService;

    @Override
    public MinutesDto.MinutesSimple create(MinutesDto.CreateReq req) {
        return minutesWriteService.create(req);
    }

    @Override
    public List<MinutesDto.MinutesSimple> listSimple() {
        return minutesReadService.list();
    }

    @Override
    public MinutesDto.MinutesDetail getDetail(Long id) {
        return minutesReadService.getDetailById(id);
    }

    @Override
    public void update(Long id, MinutesDto.UpdateReq req) {
        minutesWriteService.update(id, req);
    }

    @Override
    public void delete(Long id) {
        minutesWriteService.hardDelete(id);
    }
}

File: service/src/main/kotlin/net/huray/backend/minuting/controller/MinutesController.kt
package net.huray.backend.minuting.controller

import net.huray.backend.http.res.DoneResult
import net.huray.backend.http.res.ListResult
import net.huray.backend.http.res.SingleResult
import net.huray.backend.minuting.contract.MinutesContract
import net.huray.backend.minuting.dto.MinutesDto
import net.huray.backend.minuting.service.MinutesService
import org.springframework.web.bind.annotation.RestController

@RestController
class MinutesController(
    private val minutesService: MinutesService
) : MinutesContract {

    override fun create(req: MinutesDto.CreateReq) =
        SingleResult(minutesService.create(req))

    override fun listSimple(req: MinutesDto.ListReq) =
        ListResult(minutesService.list())

    override fun getDetail(minutesId: Long) =
        SingleResult(minutesService.getDetailById(minutesId))

    override fun update(minutesId: Long, req: MinutesDto.UpdateReq): DoneResult {
        minutesService.update(minutesId, req)

        return DoneResult()
    }

    override fun delete(minutesId: Long): DoneResult {
        minutesService.hardDelete(minutesId)

        return DoneResult()
    }

}